opensoundscape/localization.py

```diff
@@ -109,82 +99,61 @@ def estimate_location(
-            cc_threshold: see SpatialEvent documentation
+            cc_threshold: see SpatialEvent documentation. If None, uses self.cc_threshold value
+            cc_filter: filter for generalized cross correlation, see
+                opensoundscape.signal_processing.gcc()
             min_n_receivers: if number of receivers with cross correlation exceeding
-                the threshold is fewer than this, raises InsufficientReceiversError
+                the threshold is fewer than this, estimates location None
                 instead of estimating a spatial location
 
         Returns:
-            location estimate as cartesian coordinates (x,y) or (x,y,z) (units: meters)
-
-        Raises:
-            InsufficientReceiversError if the number of receivers with cross correlation
-                maximums exceeding `cc_threshold` is less than `min_n_receivers`
+            Location estimate as cartesian coordinates (x,y) or (x,y,z) (units: meters)
 
         Effects:
             sets the value of self.location_estimate to the same value as the returned location
         """
         # if cc_threshold is None, use the value stored in the object
-        if cc_threshold is None:
+        if cc_threshold == "default":
             cc_threshold = self.cc_threshold
 
-        # perform generalized cross correlation to estimate time delays
-        # (unless values are already stored in attributes)
-        if self.tdoas is None or self.cc_maxs is None:
-            self.estimate_delays()
+        # if a bandpass_range is not passed, use the value stored in the object
+        if bandpass_range == "default":
+            bandpass_range = self.bandpass_range
 
-        # filter by cross correlation threshold, removing time delays + locations
-        # if cross correlation did not exceed a minimum value
-        # (low max cross correlation values indicate low confidence that the time
-        # delay truly represents two recordings of the same sound event)
-        tdoas = self.tdoas
-        locations = self.receiver_locations
-        receiver_files = np.array(
-            self.receiver_files
-        )  # needs to be np array to access using a boolean mask
-        if cc_threshold is not None:
-            tdoas = tdoas[self.cc_maxs > cc_threshold]
-            locations = locations[self.cc_maxs > cc_threshold]
-            receiver_files = receiver_files[
-                self.cc_maxs > cc_threshold
-            ]  # fails is receiver_files is not a np.array
+        max_delay = self.max_delay
 
-        # assert there are enough receivers remaining to localize the event
-        if len(tdoas) < min_n_receivers:
-            raise InsufficientReceiversError(
-                f"Number of tdoas exceeding cc threshold ({len(tdoas)} was fewer "
-                f"than min_n_receivers ({min_n_receivers})"
+        # If no values are already stored, perform generalized cross correlation to estimate time delays
```

Comment:
what will be None if there are insufficient receivers, in the new implementation? Add a comment describing what happens if there are insufficient receivers